### PR TITLE
Reduce bugtool memory usage

### LINKED ISSUE #17309: investigate bugtool memory usage
several users have reported that cilium pods got OOM killed while running cilium-bugtool. let's investigate if there are any quick change we can make to reduce the memory consumption of the bugtool.

### PR DESCRIPTION
**Pull request by tklauser**
See individual commits and #17309 for details.

On my test system this reduces total RSS by about one third:
    
Before:

    $ /usr/bin/time -f 'RSS=%MKB' cilium-bugtool
    RSS=63168KB

After:

    $ /usr/bin/time -f 'RSS=%MKB' cilium-bugtool
    RSS=42752KB

Fixes #17309

### CODE REVIEW

File: bugtool/cmd/helper.go [hunk 1]
```diff
@@ -167,8 +167,8 @@ func hashEncryptionKeys(output string) string {
 	// Search for lines containing encryption keys.
 	// Note that `auth-trunc` is also a relevant pattern, but we already match on
 	// the more generic `auth` pattern.
+	r := regexp.MustCompile("(auth|enc|aead|comp)(.*[[:blank:]](0[xX][[:xdigit:]]+))?")
```
Comment (rolinh) [inline bugtool/cmd/helper.go]: nit: Given that `hashEncryptionKeys` itself is called in a loop, would it make sense to extract it globally?

**Review by rolinh — APPROVED** [review]
:rocket:

File: bugtool/cmd/helper.go (hunk 1 above)
Comment (tklauser) [reply bugtool/cmd/helper.go]: Good point, will move it to a global var.

**Review by michi-covalent — APPROVED** [review]

**Review by christarazi — COMMENTED** [review]
Amazing, thanks!

**Pull request merged**